### PR TITLE
Fix cycle_layout for omitted layout names

### PR DESCRIPTION
**Pull request by t-wissmann**
If a custom list of layouts is specified for cycle_layout and the
current layout is not part of this list, then cycle_layout has to pick
the first layout in the list according to the docs. The present commit
re-establishes this behaviour and adds test cases for it.

### CODE REVIEW

File: src/frametree.cpp [hunk 1]
```diff
@@ -726,7 +726,12 @@ int FrameTree::cycleLayoutCommand(Input input, Output output) {
         string curname = Converter<LayoutAlgorithm>::str(cur_frame->getLayout());
         size_t count = input.end() - input.begin();
         auto curposition = std::find(input.begin(), input.end(), curname);
-        size_t idx = MOD((curposition - input.begin()) + delta, count);
+        size_t idx = 0;  // take the from the list per default
```
Comment (bew) [inline src/frametree.cpp:729]: Missing word in the comment?

Comment (t-wissmann) [reply src/frametree.cpp]: Right, thanks! I've fixed it.

File: tests/test_layout.py [hunk 2]
```diff
@@ -813,3 +813,40 @@ def test_resize_clamp_argument_bigger(hlwm):
...
+
+
+@pytest.mark.parametrize("delta", ['-1', '+1'])
+def test_cycle_layout_name_not_in_list(hlwm, delta):
+    hlwm.call('set_layout vertical')
+
+    # if the current name is not contained in the custom list
+    # of layouts, then the first entry in the custom list
+    # has to be picked
+    hlwm.call('cycle_layout {} max grid'.format(delta))
```
Comment (blarz) [inline tests/test_layout.py:831]: Use f-string here?

File: tests/test_layout.py [hunk 3]
```diff
@@ -813,3 +813,40 @@ def test_resize_clamp_argument_bigger(hlwm):
...
+    assert current_layout_name(hlwm) == 'max'
+
+
+@pytest.mark.parametrize("delta", ['-1', '+1'])
+def test_cycle_layout_name_in_the_list(hlwm, delta):
+    hlwm.call('set_layout vertical')
+    layouts = ['vertical', 'max', 'grid']
+    # here, we omit 'horizontal' to verify that we never reach
+    # the 'horizontal' layout
+    command = ['cycle_layout', delta] + layouts
```
Comment (blarz) [inline tests/test_layout.py:842]: Call it cycle_command maybe for more clarity

File: tests/test_layout.py [hunk 4]
```diff
@@ -813,3 +813,40 @@ def test_resize_clamp_argument_bigger(hlwm):
...
+    hlwm.call('cycle_layout {} max grid'.format(delta))
+
+    assert current_layout_name(hlwm) == 'max'
+
+
+@pytest.mark.parametrize("delta", ['-1', '+1'])
+def test_cycle_layout_name_in_the_list(hlwm, delta):
+    hlwm.call('set_layout vertical')
+    layouts = ['vertical', 'max', 'grid']
+    # here, we omit 'horizontal' to verify that we never reach
```
Comment (blarz) [inline tests/test_layout.py:840]: This comment confused me. How do verify this in this test?

File: NEWS [hunk 5]
```diff
@@ -13,6 +13,8 @@ Current git version
     clients ('floatplacement=center') or leaves the floating position as is
     ('floatplacement=none')
   * New rule condition 'pgid'
+  * Bug fixes:
+    - Fix wrong behaviour in 'cycle_layout'
```
Comment (blarz) [inline NEWS:17]: Could me a bit more verbose (without reading the corresponding commit message I wouldn't know what happened here).

File: tests/test_layout.py (hunk 2 above)
Comment (t-wissmann) [reply tests/test_layout.py]: If you prefer.

File: tests/test_layout.py (hunk 3 above)
Comment (t-wissmann) [reply tests/test_layout.py]: I called it `cycle_layout_cmd`.

File: tests/test_layout.py (hunk 4 above)
Comment (t-wissmann) [reply tests/test_layout.py]: I clarified the comment. 'horizontal' is cleary not contained in the 'expected_layouts' list, so we implicitly verify that we never reach the 'horizontal' layout.

File: NEWS (hunk 5 above)
Comment (t-wissmann) [reply NEWS]: I don't know how it helps, since the bug is fixed now. But I clarified it a bit.

File: tests/test_layout.py (hunk 2 above)
Comment (blarz) [reply tests/test_layout.py]: Thanks!

File: tests/test_layout.py (hunk 3 above)
Comment (blarz) [reply tests/test_layout.py]: Thanks!

File: tests/test_layout.py (hunk 4 above)
Comment (blarz) [reply tests/test_layout.py]: Ah I see, thanks for clarifying!

File: NEWS (hunk 5 above)
Comment (blarz) [reply NEWS]: Thank you!

**Pull request merged**